eureka2-client/src/main/java/com/netflix/eureka2/client/registration/EurekaRegistrationClientImpl.java

```diff
@@ -31,17 +31,17 @@
     private final int retryWaitMillis;
 
     @Inject
-    public RegistrationHandlerImpl(ChannelFactory<RegistrationChannel> channelFactory) {
+    public EurekaRegistrationClientImpl(ChannelFactory<RegistrationChannel> channelFactory) {
         this(channelFactory, DEFAULT_RETRY_WAIT_MILLIS);
     }
 
-    /*visible for testing*/ RegistrationHandlerImpl(ChannelFactory<RegistrationChannel> channelFactory, int retryWaitMillis) {
+    /*visible for testing*/ EurekaRegistrationClientImpl(ChannelFactory<RegistrationChannel> channelFactory, int retryWaitMillis) {
         this.retryableConnectionFactory = new RetryableConnectionFactory<>(channelFactory);
         this.retryWaitMillis = retryWaitMillis;
     }
 
     @Override
-    public RegistrationResponse register(Observable<InstanceInfo> instanceInfoStream) {
+    public RegistrationRequest connect(Observable<InstanceInfo> instanceInfoStream) {
```

Comment:
I find RegistrationRequest name not fitting well here. RegistrationObservable is more transparent to the purpose of this object.